server/src/main/java/keywhiz/service/providers/ClientAuthenticator.java

```diff
@@ -120,6 +122,10 @@ public ClientAuthenticator(
   }
 
   static Optional<String> getClientName(Principal principal) {
+    if (principal instanceof SpiffePrincipal) {
```

Comment:
nit: Since we control both `SpiffePrincipal` and `CertificatePrincipal`, we could get rid of a lot of `instanceof` checks by having both principals declare a `getClientName` method and `getSpiffeId` method (maybe have a `KeywhizPrincipal` interface that they both extend which adds those two methods). Again, though, that should be a separate commit and possibly another PR.